frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx

```diff
@@ -1,44 +1,45 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { Helmet } from 'react-helmet';
 import { RouteComponentProps } from 'react-router';
 import { useTranslation } from 'react-i18next';
 import { PageBody } from '@console/shared';
 import { LoadingInline, PageHeading } from '@console/internal/components/utils';
-import { K8sResourceKind } from '@console/internal/module/k8s';
-import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
 import NamespacedPage, {
   NamespacedPageVariants,
 } from '@console/dev-console/src/components/NamespacedPage';
 import { QUERY_PROPERTIES } from '@console/dev-console/src/const';
 import ConnectedEventSource from './EventSource';
 import EventSourceAlert from './EventSourceAlert';
-import {
-  addCamelKameletSourceList,
-  useEventSourceList,
-} from '../../utils/create-eventsources-utils';
+import { useEventSourceList, useKameletSourceList } from '../../utils/create-eventsources-utils';
 import { isDynamicEventSourceKind } from '../../utils/fetch-dynamic-eventsources-utils';
-import { CamelKameletBindingModel, CamelKameletModel } from '../../models';
+import { CamelKameletBindingModel } from '../../models';
 
 type EventSourcePageProps = RouteComponentProps<{ ns?: string }>;
 
 const EventSourcePage: React.FC<EventSourcePageProps> = ({ match, location }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
-  const eventSourceList = useEventSourceList(namespace);
+  const eventSourceListData = useEventSourceList(namespace);
```

Comment:
we don't need to fetch all the event sources and perform an access check for all of them. Just get the data for the needed event source using the kind from the URL.
Icon, Description can be taken from their separate utilities